### ISSUE TITLE
Remove Excused employees from Goals, Conversations, and Shared reports

### ISSUE THREAD

**Issue by Travis-A-Clark** [bcgov/performance#773]
When an employee has been excused, they should be removed from in-app statistics and reports. Those that are manually excused will show as "active" but will have a date range entered to show when they are to be excused from the system. Those that are automatically excused will as on leave (leave of absence, leave with pay, etc.) but will not have a date range entered. Both should be removed from all stats and reports (except the Excused report itself).
<br>

**Comment by jp-Telus** [comment]
@Travis-A-Clark 

Latest status: Work in progress. ETA, early next week. I have to changed on page under My team, HR Admin and Sys Admin.

**Comment by jp-Telus** [comment]
Work Done.
additional requirement, based on the employee status to determine whether the employee is excused or not (employee_status <> 'A'), for those  excused employee :
1.  excluded from all stats and reports (except the Excused report itself).
2. include those excused employees in the count under the excuse tab.

Test on the DEV region and ready for TST migration.

**Comment by jp-Telus** [comment]
Work Done.
additional requirement, based on the employee status to determine whether the employee is excused or not (employee_status <> 'A'), for those  excused employee :
1.  excluded from all stats and reports (except the Excused report itself).
2. include those excused employees in the count under the excuse tab.

Test on the DEV region and ready for TST migration.

**Comment by jp-Telus** [comment]
Migrated to TEST region

**Comment by Travis-A-Clark** [comment]
@jp-Telus Just a heads up that the Excused Report for the Sys Admin is also off. It shows basically everyone as excused.

![image.png](https://images.zenhubusercontent.com/5ffe1f472e351dd9b643025b/20777182-9915-4f48-a76e-e42e3859275f)

**Comment by jp-Telus** [comment]
@Travis-A-Clark @steffipravasi 

Yes, the data was incorectly calculated on the excused tag, I have fixed the issue and tested on DEV region.

**Comment by Travis-A-Clark** [comment]
@jp-Telus It looks like since the work has been done to remove the Excluded employees from the reports that the Goal Summary tab no longer loads for supervisor, HR Admin, or Sys Admin. It just times out. All the other tabs load as expected.

**Comment by Travis-A-Clark** [comment]
@jp-Telus It looks like since the work has been done to remove the Excluded employees from the reports that the Goal Summary tab no longer loads for supervisor, HR Admin, or Sys Admin. It just times out. All the other tabs load as expected.

**Comment by steffipravasi** [comment]
@Travis-A-Clark I will create a new ticket for the Goals Summary issue and assign it to Dennis as James is on vacation.

**Comment by steffipravasi** [comment]
The issue of excused employees count displaying different has been resolved. The Goals Summary tab issue will be tracked in ticket #786. This ticket is ready to be moved for PO review.

**Comment by Travis-A-Clark** [comment]
@jp-Telus I am moving this ticket back to "in progress" because we will have to test this again once Dennis' new excused status is in place in the test environment.

**Comment by jp-Telus** [comment]
@Travis-A-Clark 

The following programs were updated to use the new conversation due and excused status stored in employee_demo_jr table 
1. My Team - statictics and reporting pages
2. HR Admin - statictics and reporting pages
3. System Admin - statictics and reporting pages

Note: The above programs works fine in DEV regions but the Goal page in Sys Admin page getting 504 Time Out issue in TEST region, requesting the database performance checking or tuning since we found out that double time on the SQL query under TEST region.

Thanks.

**Comment by Travis-A-Clark** [comment]
@jp-Telus The performance issues in TEST were also noted in ticket #786. Loading the Goals Summary report for any user risks timing out. Will need to resolve before we can push updates to the reporting pages.

**Comment by jp-Telus** [comment]
The 504 issue was resolved by keeping the next_conversation_date and due_date_paused fields in the users table to avoid slow down when joining to employee_demo_jr table.

The addition SQL update was located at the last step of the CalcNextConversioinDate process which was ran (at 2:00am) before notifyConversationDue processs (2:30am)

![image.png](https://images.zenhubusercontent.com/60e73283e3a918390563e770/f64cca7f-7217-48fb-bbb3-a6b8625ac698)

**Comment by steffipravasi** [comment]
Test fail
Env: Test
User: Shannon Baskerville
The team members list for Shannon with the excused employees differs from the Statistics and reports page. James looking into that issue.
![Screen Shot 2022-11-01 at 12.06.25 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/0f2e968b-2b13-4622-b550-2488235bf18a) 
![Screen Shot 2022-11-01 at 12.35.48 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/32180f60-935b-4d62-80bf-5c128c48686b)
![Screen Shot 2022-11-01 at 12.36.53 PM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/28c55c44-30a3-47b0-96b1-8dcc394eef27)

**Comment by jp-Telus** [comment]
@steffipravasi 

The issue was resolved by remove the checking on the excused_start_date and excused_end_date in users table, confirmed those fields are not in use anymore since the excuse design. The data should be in sync with the my team page. The latest changes were deployed on DEV and TEST regions,. You could proceed your testing.

Note, 3 areas were updated 
1.  Statistics Reporting under My Team
2.  Statistics Reporting under HR Admin
3.  Statistics Reporting under System Admin

Thank you.

**Comment by jp-Telus** [comment]
@Travis-A-Clark 

Per discuss, new #833 was created for the "The Open and Completed conversation reports is showing wrong numbers." issue. 

Thank you
